lib/tapioca/generator.rb

```diff
@@ -559,6 +565,31 @@ def compile_gem_rbi(gem)
       end
     end
 
+    sig { params(gem: Gemfile::GemSpec, rbi: RBI::Tree).returns(RBI::Tree) }
+    def merge_with_exported_rbi(gem, rbi)
+      return rbi unless gem.has_exported_rbi_files?
+      rbi_merge_result = gem.exported_rbi_tree
+      copied_rbi = rbi_merge_result.tree
+
+      rbi_merge_result.conflicts.each do |conflict|
+        say("\n\n  #{conflict}", :yellow)
+        say("  Found at:", :yellow)
+        say("    #{conflict.left.loc}", :yellow)
+        say("    #{conflict.right.loc}", :yellow)
+      end
+      display_conflict_warnings(rbi, copied_rbi)
+
+      RBI::Rewriters::Merge.merge_trees(rbi, copied_rbi, keep: RBI::Rewriters::Merge::Keep::LEFT)
+    end
+
+    sig { params(generated_rbi: RBI::Tree, copied_rbi: RBI::Tree).void }
+    def display_conflict_warnings(generated_rbi, copied_rbi)
+      rewriter = RBI::Rewriters::Merge.new
+      rewriter.merge(generated_rbi)
+      conflicts = rewriter.merge(copied_rbi)
+      say(conflicts.join("\n"), :yellow) unless conflicts.empty?
```

Comment:
The fact that this and the method above now report conflicts in a way that is formatted differently and containing different information needs to be cleaned up certainly.
I'm away for a while so left this commit as a WIP.